Tests/Client/Unit Tests/WebSocket/WebSocketProviderMock.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  WebSocketProviderMock.swift
+//  StreamChatClientTests
+//
+//  Created by Alexey Bukhtin on 01/05/2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import XCTest
+@testable import StreamChatClient
+
+final class WebSocketProviderMock: WebSocketProvider {
+    let request: URLRequest
+    var isConnected: Bool = false
+    let callbackQueue: DispatchQueue
+    weak var delegate: WebSocketProviderDelegate?
+    private var connectionId: String?
+    
+    var failNextConnectCount = 0
+    let timeout = 3
+    private var timer: DispatchSourceTimer?
+    
+    init(request: URLRequest, callbackQueue: DispatchQueue) {
+        self.request = request
+        self.callbackQueue = callbackQueue
+    }
+    
+    func connect() {
+        if failNextConnectCount > 0 {
+            failNextConnectCount -= 1
+            
+            DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(100)) {
+                self.delegate?.websocketDidDisconnect(self, error: nil)
+            }
+            
+            return
+        }
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(100)) {
+            self.isConnected = true
+            self.startTimer()
+            self.delegate?.websocketDidConnect(self)
+            
+            // Send the first event with the current user.
+            let event = """
+                {
+                "created_at" : "2020-05-02T13:21:03.862065063Z",
+                "me" : {
+                "id" : "steep-moon-9",
+                "banned" : false,
+                "unread_channels" : 0,
+                "mutes" : [],
+                "last_active" : "2020-05-02T13:21:03.849219Z",
+                "created_at" : "2019-06-05T15:01:52.847807Z",
+                "devices" : [],
+                "invisible" : false,
+                "unread_count" : 0,
+                "channel_mutes" : [],
+                "image" : "https://i.imgur.com/EgEPqWZ.jpg",
+                "updated_at" : "2020-05-02T13:21:03.855468Z",
+                "role" : "user",
+                "total_unread_count" : 0,
+                "online" : true,
+                "name" : "steep-moon-9",
+                "test" : 1
+                },
+                "type" : "health.check",
+                "connection_id" : "
```

Comment:
I personally prefer having this in Swift's `[String: Any]` instead of a raw string. I find it more flexible and editable (for example when you need to modify the data). But I'm OK with this too 👍

Reply:
Got it! Here I no need to reuse or mutate the data. I like your suggestion.